pkg/tekton/metapipeline/metaclient/metaclient.go

```diff
@@ -0,0 +1,324 @@
+package metaclient
```

Comment:
I don't think this additional pkg is needed. I'd just add _client.go_ to the parent package. It is also a bit too much _meta_ in this path - metapipeline/metaclient/metaclient.go

Reply:
As with jxfactory it worries me that this is duplicating the code.

Reply:
I added jxfactory as we have other uses for it outside of metaclient (e.g. for JXUI) so its a handy way to have factory methods to create kube / jx / tekton clients.

We can easily refactor the clients package to reuse jxfactory

Reply:
BTW actually there are a few changes with jxfactory to remove the CLI handling code loading kubernetes configurations; so its probably best if we have a minor bit of duplication for now for loading the kube configs between the CLI + the jxfactory. The factory code is fairly simple though; sometimes a small bit of duplication is OK